### PR TITLE
Refactor and revise Python module name

### PR DESCRIPTION
**Pull request by ExplorerRay**
As title

### CODE REVIEW

File: include/ModelAnalysis.hpp [hunk 1]
```diff
@@ -8,12 +8,19 @@
...
+  std::vector<std::vector<int64_t>> input_shapes;
+  std::vector<int64_t> output_shape;
+  std::unordered_map<std::string, size_t> ndname_to_size;
 };
 
 int64_t get_prod(std::vector<int64_t> &vec);
+NodeAnalArgs get_anal_args(onnx::NodeProto &node, const std::unordered_map<std::string, std::vector<int64_t>> &ndname_to_shape, const std::unordered_map<std::string, size_t> &ndname_to_dtype_size);
 
 AnalyzeData analyze_node_Conv(onnx::NodeProto &node, std::vector<std::vector<int64_t>> &input_shapes, std::vector<int64_t> &output_shape, std::unordered_map<std::string, size_t> &ndname_to_size);
 AnalyzeData analyze_node_Relu(onnx::NodeProto &node, std::vector<std::vector<int64_t>> &input_shapes, std::vector<int64_t> &output_shape, std::unordered_map<std::string, size_t> &ndname_to_size);
```
Comment (ThreeMonth03) [inline include/ModelAnalysis.hpp:26]: using alias declaration to avoid long lines.

File: include/InferShape.hpp [hunk 2]
```diff
@@ -9,32 +9,32 @@
 
 class InferShapeImpl {
 public:
-    InferShapeImpl() = default;
-    InferShapeImpl(const onnx::GraphProto &in_graph) : graph(in_graph) {}
+  InferShapeImpl() = default;
+  InferShapeImpl(const onnx::GraphProto &in_graph) : graph(in_graph) {}
```
Comment (ThreeMonth03) [inline include/InferShape.hpp:14]: Consider the rule of five in C++.

File: src/ModelAnalysis.cpp [hunk 3]
```diff
@@ -2,227 +2,183 @@
...
+  else if (node.op_type() == "Flatten") {
+    data = analyze_node_Flatten(node, anal_args.input_shapes, anal_args.output_shape, anal_args.ndname_to_size);
+  }
+  else if (node.op_type() == "Gemm") {
+    data = analyze_node_Gemm(node, anal_args.input_shapes, anal_args.output_shape, anal_args.ndname_to_size);
+  }
+  else {
+    std::cerr << "Error: " << node.op_type() << " not supported now\n";
+    exit(1);
+  }
```
Comment (ThreeMonth03) [inline src/ModelAnalysis.cpp:181]: Consider `enum class` to pass the type.

File: src/main.cpp [hunk 4]
```diff
@@ -5,7 +5,7 @@
 
 namespace py = pybind11;
 
-PYBIND11_MODULE(_onnxinfo, m) {
+PYBIND11_MODULE(onnxinfo, m) {
   m.doc() = "pybind11 onnxinfo module"; // optional module docstring
 
   m.def("read_onnx", &read_onnx, "A C++ function that read ONNX model");
```
Comment (ThreeMonth03) [inline src/main.cpp:11]: The file name might call `pybind.cpp`

File: src/utils.cpp [hunk 5]
```diff
@@ -35,21 +35,21 @@ void print_dim(const ::onnx::TensorShapeProto_Dimension &dim) {
...
-    std::string str = "[";
-    for (size_t i = 0; i < dims.size() - 1; ++i) {
-        str += std::to_string(dims[i]) + ", ";
-    }
-    str += std::to_string(dims.back()) + "]";
+  std::string str = "[";
+  for (size_t i = 0; i < dims.size() - 1; ++i) {
+    str += std::to_string(dims[i]) + ", ";
+  }
+  str += std::to_string(dims.back()) + "]";
```
Comment (ThreeMonth03) [inline src/utils.cpp:50]: Consider using `ofstream` to prevent programming the similar function.

File: include/ModelAnalysis.hpp [hunk 6]
```diff
@@ -8,12 +8,19 @@
 #define DIV_MACS 4
```
Comment (ThreeMonth03) [inline include/ModelAnalysis.hpp:8]: Use `constexpr int DIV_MACS = 4`.  `#define` is not modern c++.

**Review by ThreeMonth03 — COMMENTED** [review]
These suggestions may make your code clean.
By the way, maybe you should present making issues, pulling requests and cleaning codes.

File: src/ModelAnalysis.cpp (hunk 3 above)
Comment (ExplorerRay) [reply src/ModelAnalysis.cpp:181]: Is enum good with string? I found that most people use some ways to convert them, then I don't want to this which is like a workaround.

Comment (ThreeMonth03) [reply src/ModelAnalysis.cpp:181]: My fault. `std::string` is not the integral_type in cpp, so it is not suitable for replacing string with enum class.

**Pull request merged**